### PR TITLE
Sg core development

### PR DESCRIPTION
**Pull request by mgirgisf**
this pull request cover this 2 points :
1-configure and deploy sg-core
2-modify running ceilometer-agent-notification configuration to use appropriate publisher on all pipeline sinks to point to deployed sg-core

Note : in case of deployment on Standalone env the test_config file that we pass it should contains
prometheus_destination : standalone
ceilometer_pipeline_path: "/var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml"

### CODE REVIEW

File: roles/osp_observability/defaults/main.yml [hunk 1]
```diff
@@ -5,3 +5,9 @@ workdir: /var/lib/osp-observability
 
 openstack_cloud_name: overcloud
 prometheus_scrape_interval: "1s"
+
+#sg-vars
+ceilometer_port: '6420'
```
Comment (csibbitt) [inline roles/osp_observability/defaults/main.yml:10]: The naming of this variable tripped me up. 

`prometheus_port` is the port prometheus is listening on, so I would expect`ceilometer_port` is the port ceilometer is listening on. Instead I think this is the port that _sg_core is listening on_ for client connection from ceilometer.

Suggest `sg_ceilometer_port`

File: roles/osp_observability/tasks/configure_ceilometer.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+- name: check if ceilometer service is deployed
+  fail:
+    msg: Invalid Env, it doesn't contain ceilometer.
+  when: ansible_facts.services["tripleo_ceilometer_agent_notification.service"] is not defined
+
+- name: Create ceilometer pipeline.yml file
+  template:
+    src: ceilometer_pipeline.yml.j2
+    dest: "{{ ceilometer_pipeline_path }}"
```
Comment (csibbitt) [inline roles/osp_observability/tasks/configure_ceilometer.yml:13]: Is it okay for us to override the entire contents of this file? Could it be that users have configured ceilometer pipelines via tripleo that we should not be clobbering?

**Review by csibbitt — APPROVED** [review]
This looks like a good start to me and I'm okay to merge it if it's working, but I am concerned about taking full control of the pipeline.yml and have a minor comment about variable naming clarity.

File: roles/osp_observability/tasks/configure_ceilometer.yml (hunk 2 above)
Comment (paramite) [reply roles/osp_observability/tasks/configure_ceilometer.yml:13]: Agree, we should just switch the sinks to Jaromir's TCP publisher. Will work on that.

File: roles/osp_observability/defaults/main.yml (hunk 1 above)
Comment (paramite) [reply roles/osp_observability/defaults/main.yml:10]: Agree.

**Comment by paramite** [comment]
I will just merge this now because I cannot modify Marihan's PR and continue with new PR.

**Pull request merged**